code-challenges.test.js

```diff
@@ -35,22 +87,94 @@ var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+    let leastToGreatest = oddNumArr.sort((a, b) => a - b);
+    //return the values
+    return leastToGreatest
+}
+//console.log(oddFilter(fullArr1));
+// -- Got the green --
+
+// --------------------3) Create a function that takes in an array and returns an array of the 
+// accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
-
-var numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
+//                   0           1                       2       3
+var numbersToAdd1 = [2,         4,                     45,      9]
+//                 0+oddindex0   newindex0+oddindex1     
+// Excpected output:[2,          6,                     51,     60]
 
 var numbersToAdd2 = [0, 7, -8, 12]
 // Expected output: [0, 7, -1, 11]
 
 var numbersToAdd3 = []
 // Expected output: []
-
-
+// a describe method that lists the name of the function OR naming of the particular test.
+describe("twistPlus", () => {
+    var numbersToAdd1 = [2, 4, 45, 9]
+    var numbersToAdd2 = [0, 7, -8, 12]
+    var numbersToAdd3 = []
+
+    // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+    it("with expect statements for each of the variables provided.var numbersToAdd1 = [2, 4, 45, 9] ===> Excpected output: [2, 6, 51, 60]", () => {
+
+        //an expect method, nested within the test block, calling on the oddFilter() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+        expect(twistPlus(numbersToAdd1)).toEqual([2, 6, 51, 60])
+        expect(twistPlus(numbersToAdd2)).toEqual([0, 7, -1, 11])
+        expect(twistPlus(numbersToAdd3)).toEqual([])
+    })
+})
+// -- Got the red --  
 // b) Create the function that makes the test pass.
+// create a function 
+const twistPlus = (array) => {
+    // declare the an empty arrays
+    var newArray = []
+    //if statement starts, if the array from the argument is empty,which the length is 0
+    if (array.length === 0) {
+        //we can return the newArray,since the newArray is empty
+        return newArray;
+        
+    }
+    //if it's not empty 
+    else {
+        //first we push the first value of the argument array to the newArray
+        //newArray[0] = array[0]
+        newArray.push(array[0])
+        //for loop starts, search the argument array starts with sencond value of the array
+        for (let i = 1; i < array.length; i++) {
+            //let the array[i] + newArray[i - 1] then push to newArray
+            //newArray[1] = array[1] + newArray[0]
+            //newArray[2] = array[2] + newArray[1]
+            //newArray[3] = array[3] + newArray[2]
+            //..
+            //newArray[i] = array[i] + newArray[i-1]
+            newArray.push(array[i] + newArray[i - 1])
+        }
+        //return newArray
+        return newArray;
+    }
+}
+// -- Got the green --  
```

Comment:
Logic looks good. I think in this case you could skip the if logic as your else statement would still return an empty array if given an empty array as an input.